src/Evolution/Systems/GeneralizedHarmonic/Equations.cpp

```diff
@@ -447,45 +446,47 @@ void UpwindFlux<Dim>::operator()(
 }
 
 template <size_t Dim>
-typename Tags::CharacteristicFields<Dim, Frame::Inertial>::type
+db::item_type<Tags::CharacteristicFields<Dim, Frame::Inertial>>
 UpwindFlux<Dim>::weight_char_fields(
-    const typename Tags::CharacteristicFields<Dim, Frame::Inertial>::type&
+    const db::item_type<Tags::CharacteristicFields<Dim, Frame::Inertial>>&
         char_fields_int,
-    const typename Tags::CharacteristicSpeeds<Dim, Frame::Inertial>::type&
+    const db::item_type<Tags::CharacteristicSpeeds<Dim, Frame::Inertial>>&
         char_speeds_int,
-    const typename Tags::CharacteristicFields<Dim, Frame::Inertial>::type&
+    const db::item_type<Tags::CharacteristicFields<Dim, Frame::Inertial>>&
         char_fields_ext,
-    const typename Tags::CharacteristicSpeeds<Dim, Frame::Inertial>::type&
+    const db::item_type<Tags::CharacteristicSpeeds<Dim, Frame::Inertial>>&
         char_speeds_ext) const noexcept {
-  const auto u_psi_int = get<Tags::UPsi<Dim, Frame::Inertial>>(char_fields_int);
-  const auto u_zero_int =
+  const auto& u_psi_int =
+      get<Tags::UPsi<Dim, Frame::Inertial>>(char_fields_int);
+  const auto& u_zero_int =
       get<Tags::UZero<Dim, Frame::Inertial>>(char_fields_int);
-  const auto u_plus_int =
+  const auto& u_plus_int =
       get<Tags::UPlus<Dim, Frame::Inertial>>(char_fields_int);
-  const auto u_minus_int =
+  const auto& u_minus_int =
       get<Tags::UMinus<Dim, Frame::Inertial>>(char_fields_int);
 
-  const Scalar<DataVector> char_speed_u_psi_int{char_speeds_int[0]};
-  const Scalar<DataVector> char_speed_u_zero_int{char_speeds_int[1]};
-  const Scalar<DataVector> char_speed_u_plus_int{char_speeds_int[2]};
-  const Scalar<DataVector> char_speed_u_minus_int{char_speeds_int[3]};
+  const DataVector& char_speed_u_psi_int{char_speeds_int[0]};
```

Comment:
We don't know what the behaviour for this constructor is and think that instead the assignment operator should be used.

Reply:
They're both the same. The assignment operator is never used, it's always the constructor, even if you type `=`. You can't assign to something that doesn't exist, but you can construct it.